flight/Libraries/math/pid.c

```diff
@@ -199,6 +230,67 @@ void pid_configure(struct pid *pid, float p, float i, float d, float iLim)
 }
 
 /**
+ * Assigns a pid_deadband struct to the PID controller.
+ * @param[out] pid The PID structure to configure
+ * @param[in] deadband The external deadband structure to assign
+ */
+void pid_assign_deadband(struct pid *pid, struct pid_deadband *db)
+{
```

Comment:
We can't free anything.

I'm just saying-- shouldn't the rate controller always have the deadband if configured?  There is no need to "take it out"

It looks like rihgt now you have no deadband on the inner loop for attitude mode.  Why not?  If the deadband is to prevent ringing in the inner loop, doesn't that hold true whether stick input or the attitude controller is driving it?